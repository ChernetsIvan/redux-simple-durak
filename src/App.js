import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {clickOnStartGameButton, clickOnBeginGameButton, clickOnRadioWhoMoveFirst} from './actions';

import GameScreen from './components/main/GameScreen';
import StartScreen from './components/main/StartScreen';

class App extends Component {
    render() {
        console.log(this.props);

        let output = null;
        if(this.props.isRenderSettingsForStartNewGame){
            output = 
                <StartScreen 
                    isFirstMovePlayer={this.props.isFirstMovePlayer}
                    onClickStartGame={this.props.onStartGameClick}
                    onChangeRadio={this.props.onRadioChange} />
        }else{
            output = 
                <GameScreen 
                    handleBeginGameClick={this.props.onBeginGameClick}
                    gameMode={this.props.gameMode}
                    computerCards={this.props.computerCards}
                    playerCards={this.props.playerCards}
                    fullDeck={this.props.fullDeck}
                    firstStart={this.props.firstStart} 
                    aiField={this.props.aiField}
                    playerField={this.props.playerField}/>
        }
        return (
            <div>
                {output}
            </div>
        );
    }
}

App.propTypes = {
    isRenderSettingsForStartNewGame: PropTypes.bool.isRequired,
    trumpSuit: PropTypes.object.isRequired,    
    fullDeck: PropTypes.array.isRequired,
    computerCards: PropTypes.array.isRequired,
    aiField: PropTypes.array.isRequired,
    playerField: PropTypes.array.isRequired,
    playerCards: PropTypes.array.isRequired,
    playerStartInd: PropTypes.number.isRequired,
    playerEndInd: PropTypes.number.isRequired,
    firstStart: PropTypes.bool.isRequired,
    isFirstMovePlayer: PropTypes.bool.isRequired,
    gameMode: PropTypes.string.isRequired,

    onBeginGameClick: PropTypes.func.isRequired,
    onStartGameClick: PropTypes.func.isRequired,
    onRadioChange: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
    isRenderSettingsForStartNewGame: state.startGame.isRenderSettingsForStartNewGame,
    trumpSuit: state.startGame.trumpSuit,
    fullDeck: state.startGame.fullDeck,
    computerCards: state.startGame.computerCards,
    aiField: state.startGame.aiField,
    playerField: state.startGame.playerField,
    playerCards: state.startGame.playerCards,
    playerStartInd: state.startGame.playerStartInd,
    playerEndInd: state.startGame.playerEndInd,
    firstStart: state.startGame.firstStart,
    isFirstMovePlayer: state.startGame.isFirstMovePlayer,
    gameMode: state.startGame.gameMode
});


const mapDispatchToProps =  dispatch => ({    
    onStartGameClick: () => {    
        dispatch(clickOnStartGameButton());
    },
    onBeginGameClick: () => {
        dispatch(clickOnBeginGameButton());
    },
    onRadioChange: (isFirstMovePlayer) => {
        dispatch(clickOnRadioWhoMoveFirst(isFirstMovePlayer));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
