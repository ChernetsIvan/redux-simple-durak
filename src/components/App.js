import React from "react";
import PropTypes from "prop-types";

import StartScreenContainer from "./../containers/StartScreen/StartScreenContainer";
import GameScreenContainer from "./../containers/GameScreen/GameScreenContainer";

const App = ({ showStartScreen }) =>
  showStartScreen ? <StartScreenContainer /> : <GameScreenContainer />;

export default App;

App.propTypes = {
  showStartScreen: PropTypes.bool.isRequired
};
