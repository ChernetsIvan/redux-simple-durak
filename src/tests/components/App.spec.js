import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-15";
import toJson from 'enzyme-to-json';

import App from "./../../components/App";

Enzyme.configure({ adapter: new Adapter() });

describe("App shallow render test", () => {
  test("App recieve showStartScreen=false", () => {
    const component = shallow(<App showStartScreen={false} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  test("App recieve showStartScreen=true", () => {
    const component = shallow(<App showStartScreen />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
