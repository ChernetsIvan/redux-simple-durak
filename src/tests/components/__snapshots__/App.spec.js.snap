// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App shallow render test App recieve showStartScreen=false 1`] = `<Connect(GameScreen) />`;

exports[`App shallow render test App recieve showStartScreen=true 1`] = `<Connect(StartScreen) />`;
